### PR TITLE
timed-independent-assessment-des-todo

### PR DESCRIPTION
**Pull request by DesTodo**
@mikedao @laurenfazah 
Spec was working fine but then I started to get an error that I can't user "scenario" although I was using it before. I tried "describe" and "it" but neither worked. Thank you!

### CODE REVIEW

File: spec/models/job_spec.rb [hunk 1]
```diff
@@ -32,5 +32,10 @@
       job = Job.new(title: "Software", level_of_interest: 70, description: "Wahooo")
       expect(job).to respond_to(:company)
     end
+
+    it "has many tags" do
+      job = Job.new(title: "Software", level_of_interest: 70, description: "Wahooo")
+      expect(job).to respond_to(:tags)
+    end
```
Comment (laurenfazah) [inline spec/models/job_spec.rb:39]: Love the use of unit tests

File: spec/models/tag_spec.rb [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+      it "has many jobs" do
+        tag = Tag.new(name: "Technology")
+        job = Job.new(title: "Developer", level_of_interest: 40, city: "Denver")
+
+        tag.jobs << job
+
+        expect(tag.jobs).to include(job)
+        expect(tag).to respond_to(:jobs)
+      end
+    end
```
Comment (laurenfazah) [inline spec/models/tag_spec.rb:29]: Great work fleshing out Tag unit tests as well.

File: app/models/tagging.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Tagging < ApplicationRecord
+  belongs_to :job
+  belongs_to :tag
+end
```
Comment (laurenfazah) [inline app/models/tagging.rb:4]: I would have loved to see you use a test to drive the development of a join table.

File: db/seeds.rb [hunk 4]
```diff
@@ -1,10 +1,12 @@
 Job.destroy_all
+Tag.destroy_all
 Company.destroy_all
 
 
 COMPANIES = ["ESPN", "Aetna", "United Airlines", "Denver Public Schools", "Shopify", "Starbucks", "Pivotal Labs", "Captain U"]
 JOBS = ["Engineering", "Development", "Dev Ops", "Quality Assurance", "Teacher", "Product Manager", "Consultant", "Community Manager"]
 CITIES = ["Seattle", "Denver", "Portland", "Indianapolis", "Madison", "Orlando", "San Diego", "Austin", "Las Vegas", "Little Rock", "Boise", "Eugene", "Oakland"]
+TAGS = ["Software", "Technology", "Healthcare", "Education", "Finance"]
```
Comment (laurenfazah) [inline db/seeds.rb:9]: What brought you here to seed tags?

File: spec/models/tag_spec.rb [hunk 5]
```diff
@@ -27,5 +27,19 @@
...
+        company = Company.create(name: "Dropbox")
+        job_1 = company.jobs.create!(title: "Developer", level_of_interest: 80, description: "DevOps experience required", city: "Denver")
+        job_2 = company.jobs.create!(title: "Developer", level_of_interest: 80, description: "IT experience required", city: "Denver")
+
+        tag = job_1.tags.create(name: "Technology")
+        expect(tag.job_count).to eq(1)
+
+        job_2.tags << tag
+        expect(tag.job_count).to eq(2)
+      end
```
Comment (laurenfazah) [inline spec/models/tag_spec.rb:42]: Great cascade of set up and expectations!

File: app/models/tag.rb [hunk 6]
```diff
@@ -7,4 +7,8 @@ class Tag < ApplicationRecord
   def job_count
     jobs.count
   end
+
+  def job_salary
+    jobs.average(:salary)
+  end
 end
```
Comment (laurenfazah) [inline app/models/tag.rb]: Where is the test for this? Otherwise, really elegant solution!

File: spec/features/jobs/user_sees_a_job_spec.rb [hunk 7]
```diff
@@ -19,16 +19,31 @@
...
     job_1.tags << tag
     job_2.tags << tag
 
     visit company_job_path(company, job_1)
 
-    expect(tag.job_count).to eq (2)
+    expect(tag.job_salary).to eq(25000)
+    expect(page).to have_content("25000")
+  end
   end
```
Comment (laurenfazah) [inline spec/features/jobs/user_sees_a_job_spec.rb]: Good updates to this feature spec.

**Comment by laurenfazah** [comment]
Des, your test-driving process was pretty great overall. There are somethings that still could have used TDD, but the tests you did write were fantastic.

Your use of ActiveRecord was very strong as well. 

The only thing I can think of to make this stronger would be adding sad path and edge case scenarios into your tests. Make sure that content that shouldn't be on the page isn't, etc.

Also, it looks like your RSpec error came from nesting a `scenario` block inside another. Be careful of this!

Great work!
